### PR TITLE
journalpump: support remote syslog

### PR DESCRIPTION
**Pull request by mteiste**

### CODE REVIEW

File: journalpump/journalpump.py [hunk 1]
```diff
@@ -3,26 +3,29 @@
...
-import kafka.common
 import logging
 import re
-import requests
 import select
 import socket
-import systemd.journal
 import time
 import uuid
+from functools import reduce
```
Comment (Ormod) [inline journalpump/journalpump.py]: Please don't reorder the imports.

File: journalpump/rsyslog.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+def _rfc5424(*, pri, timestamp, hostname, app_id, proc_id, msg_id, msg, sd):
+    data = f"<{pri}>1 {timestamp} {hostname} {app_id} {proc_id} {msg_id}"
+    if sd is not None:
+        data += f" [{sd}]"
+    data += f" {msg}\n"
+    return data.encode("utf-8", "replace")
+
+
+# pylint: disable=unused-argument
```
Comment (Ormod) [inline journalpump/rsyslog.py]: Could name this a bit more verbose than _rfc3164, something like rfc_3164_formatter or somesuch.

File: test/test_journalpump.py [hunk 3]
```diff
@@ -1,9 +1,14 @@
+import json
```
Comment (Ormod) [inline test/test_journalpump.py]: Alphabetically this would come after the from's

File: journalpump/rsyslog.py (hunk 2 above)
Comment (mteiste) [reply journalpump/rsyslog.py]: fixed

File: journalpump/journalpump.py (hunk 1 above)
Comment (mteiste) [reply journalpump/journalpump.py]: order restored

File: test/test_journalpump.py (hunk 3 above)
Comment (mteiste) [reply test/test_journalpump.py]: fixed

**Pull request merged**